test/test_image.py

```diff
@@ -325,5 +267,146 @@ def test_decode_jpeg_cuda_errors():
         torch.ops.image.decode_jpeg_cuda(data, ImageReadMode.UNCHANGED.value, 'cpu')
 
 
+@cpu_only
+def test_encode_jpeg_errors():
+
+    with pytest.raises(RuntimeError, match="Input tensor dtype should be uint8"):
+        encode_jpeg(torch.empty((3, 100, 100), dtype=torch.float32))
+
+    with pytest.raises(ValueError, match="Image quality should be a positive number "
+                                         "between 1 and 100"):
+        encode_jpeg(torch.empty((3, 100, 100), dtype=torch.uint8), quality=-1)
+
+    with pytest.raises(ValueError, match="Image quality should be a positive number "
+                                         "between 1 and 100"):
+        encode_jpeg(torch.empty((3, 100, 100), dtype=torch.uint8), quality=101)
+
+    with pytest.raises(RuntimeError, match="The number of channels should be 1 or 3, got: 5"):
+        encode_jpeg(torch.empty((5, 100, 100), dtype=torch.uint8))
+
+    with pytest.raises(RuntimeError, match="Input data should be a 3-dimensional tensor"):
+        encode_jpeg(torch.empty((1, 3, 100, 100), dtype=torch.uint8))
+
+    with pytest.raises(RuntimeError, match="Input data should be a 3-dimensional tensor"):
+        encode_jpeg(torch.empty((100, 100), dtype=torch.uint8))
+
+
+def _collect_if(cond):
+    # TODO: remove this once test_encode_jpeg_windows and test_write_jpeg_windows
+    # are removed
+    def _inner(test_func):
+        if cond:
+            return test_func
+        else:
+            return pytest.mark.dont_collect(test_func)
+    return _inner
+
+
+@cpu_only
+@_collect_if(cond=IS_WINDOWS)
+def test_encode_jpeg_windows():
+    # This test is *wrong*.
+    # It compares a torchvision-encoded jpeg with a PIL-encoded jpeg, but it
+    # starts encoding the torchvision version from an image that comes from
+    # decode_jpeg, which can yield different results from pil.decode (see
+    # test_decode... which uses a high tolerance).
+    # Instead, we should start encoding from the exact same decoded image, for a
+    # valid comparison. This is done in test_encode_jpeg, but unfortunately
+    # these more correct tests fail on windows (probably because of a difference
+    # in libjpeg) between torchvision and PIL.
+    # FIXME: make the correct tests pass on windows and remove this.
```

Comment:
I agree with this comment. The right comparison would be to encode from the same decoded image and compare the results. Unfortunately as Nicolas explains, that's going to fail on Windows. I would not be surprised if the expected image is created on Linux or macOS to make this work. Hence this test is misleading and does not test that the encoding on our side is the same as the encoding on PIL side on the same platform.

Having said that, it's good that Nicolas found a way around to maintain the test until we decide whether we want to keep it or drop it. I'm on the fence on keeping it and Nicolas briefly considered dropping it. I don't have strong opinions on this. @fmassa thoughts?

Reply:
I think the underlying issue might be that we are using different libjpeg for different OSes, which is less than ideal.

Fixing this would probably fix the issue.

Reply:
I think the key issue comes from the fact that apparently on windows, the libjpeg version of PIL is different from that of torchvision.

Having different libjpeg versions across OSes wouldn't be a problem if for a given OS, the libjpeg version was the same for both PIL and torchvision I think.

Reply:
@NicolasHug it would be good to log the libraries installed in https://github.com/pytorch/vision/blob/96ad7f0c80538e06323ab2a4d2d4fb513022f59a/.circleci/unittest/windows/scripts/setup_env.sh#L39 and https://github.com/pytorch/vision/blob/96ad7f0c80538e06323ab2a4d2d4fb513022f59a/.circleci/unittest/linux/scripts/setup_env.sh#L45 , our current constraint on libjpeg is `<=9b`, which in principle allows for different versions to be installed on different OSes, if there is one version missing in conda

Reply:
But if the problem lies within PIL, this means that if we were to add a test comparing the encoding of PIL on the CI itself it should fail?

Reply:
>  it would be good to log the libraries installed 

I'm not sure how to best do that yet but I started looking into it in https://github.com/pytorch/vision/pull/3968